src/pages/index.js

```diff
@@ -6,74 +6,134 @@ import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
 import "../pages/index.css";
 import {
-  initialCards,
+  //initialCards,
   config,
   profileEditForm,
   addCardFormElement,
   profileEditButton,
   addNewCardButton,
   profileTitleInput,
   profileDescriptionInput,
+  profileAvatarButton,
+  editAvatarForm,
+  //cardListEl,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupConfirmDelete from "../components/popupConfirmDelete.js";
+//----------------------------------API--------------------------------------//
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "096eb059-8249-432c-aed5-ea6f0d18c1d5",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+let cardSection;
+api
+  .getInitialCards()
+  .then((cardData) => {
+    cardSection = new Section(
+      {
+        items: cardData,
+        renderer: (cardData) => {
+          const cardElement = getCardElement(cardData);
+          cardSection.addItem(cardElement);
+        },
+      },
+      ".gallery__cards"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+api
+  .getUser()
+  .then((inputValues) => {
+    userInfo.setUserInfo(inputValues.name, inputValues.about);
+    userInfo.setAvatar(inputValues.avatar);
+  })
+  .catch((err) => console.error(err));
+
+function handleAddCardFormSubmit(cardData) {
+  newCardPopup.viewLoading(true);
+  api
+    .addCard(cardData)
+    .then((data) => {
+      cardSection.addItem(getCardElement(data));
+      addCardValidator.disableButton();
+      newCardPopup.resetForm();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      newCardPopup.viewLoading(false);
+    });
+}
 
 //----------------------------event handlers---------------------------------//
 function handleImageClick(cardData) {
   popupImage.open(cardData);
 }
 
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.name, inputValues.job);
-  profileEditValidator.disableButton();
-  profileEditPopup.close();
-}
-
-function handleAddCardFormSubmit(cardData) {
-  cardSection.addItem(
-    getCardElement({ name: cardData.name, link: cardData.link })
-  );
-  addCardValidator.disableButton();
-  newCardPopup.resetForm();
-  newCardPopup.close();
+  profileEditPopup.viewLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```